test/jdk/java/net/Socket/UdpSocket.java

```diff
@@ -133,8 +134,21 @@ public void testMaxSockets() throws Exception {
         }
     }
 
+
     private Socket newUdpSocket() throws IOException {
-        return new Socket(InetAddress.getLoopbackAddress(), 8000, false);
+        Socket newUdpSocket = null;
+
+        try {
+            newUdpSocket = new Socket(InetAddress.getLoopbackAddress(), 8000, false);
+        } catch (SocketException soEx) {
+            if (soEx instanceof BindException) {
+                System.out.println("BindException caught retry Socket creation");
+                newUdpSocket = new Socket(InetAddress.getLoopbackAddress(), 8000, false);
+            } else {
+                throw soEx;
+            }
+        }
+        return newUdpSocket;
```

Comment:
I added this test in advance of JEP 353 as we didn't have much coverage for this deprecated constructor. No objection to the retry if it helps.  Is the catching of SocketException a left over from testing? I assume the patch can be simplified down to:

```
try {
   return new Socket(InetAddress.getLoopbackAddress(), 8000, false);
} catch (BindException e) {
    System.out.println(...);
   return new Socket(InetAddress.getLoopbackAddress(), 8000, false);
}
```

Reply:
yes, thanks for that  ... updated as requested

Reply:
Thanks, and if you want to keep it consistent with the existing code then you could rename "Socket newUdpSocket" and "biEx", or just change it to "return new Socket(...)".

Reply:
BTW: Is one retry enough? There is at least one other replace where we've had to retry to workaround a macOS bug and one retry was enough there too.

Reply:
I have submitted a significant number of MACH5 job runs with repeat mode over the past week - approx  50 x 50,  and observed the retry to have been triggered about 5 times, this gives some level of assurance that test stability will exist. But CI builds will exercise this a lot more.

Reply:
updated as requested newUdpSocket --> s, and biEx --> unexpected